src/chapters/sql-part2/complex-queries.rst

```diff
@@ -0,0 +1,350 @@
+included in the result set.
+
+Complex queries often combine joins and subqueries, and they may also include
+multiple ``AND``/``OR`` operators within the ``WHERE`` clause.
+
+Review Joins
+------------
+
+Code along with the following examples using the ``storage`` database and
+tables you created earlier.
+
+Inner Join
+^^^^^^^^^^^
+
+Each record in ``writing_supply`` stores a ``utensil_type`` value. We could use
+a simple query to filter the records by type, but this would not contain data
+such as ``quantity``, which is stored in a different table.
+
+To return a result set that contains information that appears in both the
+``writing_supply`` and ``pencil_drawer`` tables, we use an ``INNER JOIN``.
+
+.. admonition:: Example
+
+   .. sourcecode:: SQL
+      :linenos:
+
+      SELECT *
+      FROM writing_supply
+      INNER JOIN pencil_drawer ON writing_supply.supply_id = pencil_drawer.supply_id;
+
+   The result set contains all of the records from both tables that have matching
+   ``supply_id`` values.
+
+   .. figure:: ./figures/inner-join-result-set.png
+      :alt: Result set for an inner join of the writing_supply and pencil_drawer tables.
+
+   We could accomplish the same result using a series of simple queries, but
+   this would be inefficient. We would need to first run a query on
+   ``writing_supply`` and then use the results to shape one or more queries on
+   ``pencil_drawer``.
+
+To reduce the size of the result set, we can request specific fields and add
+conditions to the query:
+
+.. admonition:: Example
+
+   .. sourcecode:: SQL
+      :linenos:
+
+      SELECT writing_supply.supply_id, pencil_type, drawer_id, quantity
+      FROM writing_supply
+      INNER JOIN pencil_drawer ON writing_supply.supply_id = pencil_drawer.supply_id
+      WHERE refill = true AND pencil_type = "Mechanical";
+
+   **Result Set**
+
+   .. figure:: ./figures/inner-join-with-conditions.png
+      :alt: Result set for an inner join with a WHERE clause.
+
+   Note that in line 1, we need to specify the source for ``supply_id``, since
+   both tables contain a column with that name.
+
+Left/Right Join
+^^^^^^^^^^^^^^^^
+
+We can use a ``LEFT`` or ``RIGHT`` join to retain all of the records from one
+table and pull in overlapping data from another.
+
+.. admonition:: Example
+
+   .. sourcecode:: SQL
+      :linenos:
+
+      SELECT writing_supply.supply_id, utensil_type, drawer_id, color
+      FROM writing_supply
+      LEFT JOIN pen_drawer ON writing_supply.supply_id == pen_drawer.supply_id;
+
+   The result set contains ``null`` values for any rows that involve pencils. The
+   left join retains all of the data in ``writing_supply``, but it can only
+   combine that information with data from ``pen_drawer`` if the rows share
+   ``supply_id`` values.
+
+   .. figure:: ./figures/left-join-result-set.png
+      :alt: Result set for an inner join with a WHERE clause.
+
+As with inner joins, we can restrict the size of the result set:
```

Comment:
I would specify that the size of the result set is decreasing because of the additional condition.

Reply:
Done.  Thanks.